src/VASSAL/counters/Footprint.java

```diff
@@ -76,6 +79,9 @@
 
   // Type Variables (Configured in Ed)
   protected NamedKeyStroke trailKey;                // Control Key to invoke
+  protected NamedKeyStroke trailKeyOn;              //BR// Control Key to force trails on
```

Comment:
I would give this field and the other new fields private visibility instead of protected - there are no classes that inherit from this in Vassal, and if there are subclasses of this in the modules they won't need these fields right away. If we make them protected now, we expose them as public API which we may have to support for a long time to keep backwards compatibility.

If at some point in future this class is inherited from and the subclass needs access to these fields, they can still be made protected, or stay private and get public getters/setters.

Reply:
I don't have major objections if that's the consensus, although it does mean that some hypothetical overriding "custom class" wouldn't have access -- I'd used the protected/private levels for other similar items in the class (trailKey that already existed was protected, so went with that for e.g. trailKeyOn). So it's a little contrary to the normal Vassal architecture though I realize there are downsides in that. (What would anybody ever want to override this for? I guess to make it so those commands could have popup menu items attached to them, which I currently haven't provided in the interests of making a relatively minimal change).   But I certainly agree it wouldn't compromise the core value of the fix. Will await further discussion.

Reply:
My inclination is to make new fields private unless and until someone asks for them to be protected.

Reply:
Fixed & pushed.

Reply:
But if we add private fields, can we please, please add public or at least protected getters and setters? The 'ask for them to be protected' process just wastes developers time and our time.

Reply:
Well, hmm. As this is a module-designer-facing class, if you want them protected, go for it.

Reply:
Okay, done. It's a new PR since this one is merged, but does only that.

Reply:
Thanks Brian. The Traits in particular are the first classes most developers want to override. There should be no private members without at least protected getters/setters in any trait. Many years back, I went and pre-emptively changed any privates I found to protected (Yes, I know Yan, should have added getters/setters, cest la vie). There may be a few more hanging round that I missed or have been added since.